spec/fileutils_spec.rb

```diff
@@ -673,6 +693,24 @@
         expect(File.directory?('/path/to/some')).to be true
       end
     end
+
+    context 'when passing options' do
+      context 'when passing mode parameter' do
+        it 'creates directory with specified permissions' do
+          described_class.mkdir_p('/test-dir', :mode => 0654)
+          expect(File.exist?('/test-dir')).to be true
+          expect(File.stat('/test-dir').mode).to eq(0100654)
+        end
+      end
+
+      context 'when passing noop parameter' do
+        it 'does not create any directories' do
+          described_class.mkdir_p(['/test-dir', '/another-dir'], :noop => true)
+          expect(File.directory?('/test-dir')).to be false
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.